### DISCUSSION TITLE
For @ bable / core, there are many empty lines after using generator to convert it into JS code

### DISCUSSION THREAD

**Discussion by Mankvis** [babel/babel#12271] [General]
1. Use parse to convert JS code into ast structure

2. Adjust the structure accordingly

3. Use generator to convert AST to JS code

Problem: a lot of new lines are found, because there are a lot of code replaced, so "retain lines" cannot be used, otherwise there will be large empty lines

**Comment by JLHwung** [comment]
Babel does not focus on preserving / removing whitespaces from the generated result. You can pass the generated code to a formatter, e.g. [`prettier`](https://prettier.io/).

**Reply by Mankvis** [reply]
Sorry, I didn't find the document using prettier to delete empty lines. Could you help me write a demo? Thank you very much.
